src/new/wallet.js

```diff
@@ -505,6 +500,9 @@ class HathorWallet extends EventEmitter {
    * @inner
    */
   getAddressAtIndex(index) {
+    if (this.preCalculatedAddresses && this.preCalculatedAddresses[index]) {
```

Comment:
Why don't we change a bit this and throw an error if you have `this.preCalculatedAddresses` but not with this index?

```
if (this.preCalculatedAddresses) {
  if (!this.preCalculatedAddresses[index]) throw Error();
  return this.preCalculatedAddresses[index];
}
```

Otherwise we would be calling a method to generate a new address but we are using pre calculated addresses

Reply:
I think we may want to start with the first 50 addresses preCalculated, then load more addresses as needed.